akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -126,6 +127,29 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         errorCode should ===(ErrorCode.COMPRESSION_ERROR)
       }
 
+      "GOAWAY when the response has headers mid stream" in new TestSetup with NetProbes with Http2FrameHpackSupport {
+        val streamId = 0x1
+        emitRequest(streamId, Get("/"))
+        expectDecodedHEADERS(streamId, endStream = true)
+
+        sendHEADERS(streamId, endStream = false, Seq(
+          RawHeader(":status", "200"),
+          RawHeader("content-type", "application/octet-stream")
+        ))
+
+        val response = expectResponse()
+        response.entity shouldBe a[Chunked]
+
+        sendDATA(streamId, endStream = false, ByteString("asdf"))
+        sendHEADERS(streamId, endStream = false, Seq(RawHeader("X-mid-stream", "badvalue")))
+
+        // FIXME need something like the Http2ServerSpec.expectWindowUpdate() here
+        expectFrameFlagsStreamIdAndPayload(FrameType.WINDOW_UPDATE)
+        
+        val (_, error) = expectGOAWAY(streamId)
+        error should ===(ErrorCode.PROTOCOL_ERROR)
```

Comment:
Also manually verified by failing this test and seeing that there is a debug text in the log mentioning mid stream headers in addition to the PROTOCOL_ERROR

Reply:
You could also use `expect[GoAwayFrame]()` if you want to assert on the debug string.

Reply:
Ah, nice, that's good to make sure the test isn't accidentally failing from abusing the protocol in some other way, I'll change.